core/templates/pages/exploration-player-page/services/audio-preloader.service.ts

```diff
@@ -93,15 +100,24 @@ export class AudioPreloaderService {
   private getAudioFilenamesInBfsOrder(sourceStateName: string): string[] {
     const languageCode = (
       this.audioTranslationLanguageService.getCurrentAudioLanguageCode());
-    const allVoiceovers = this.exploration.getAllVoiceovers(languageCode);
-    const bfsTraversalOfStates = (
-      this.computeGraphService.computeBfsTraversalOfStates(
-        this.exploration.getInitialState().name, this.exploration.getStates(),
-        sourceStateName));
+    let allVoiceovers = null;
+    if (languageCode !== null) {
```

Comment:
Is it correct if in this place `languageCode` is `null`? If not can we just throw an error?

Reply:
No, we can't throw an error, as the language code can be a null value. I was doing the same thing previously which caused all the e2e to fail at once due to the error thrown.

Reply:
What does it logically mean when it is `null`? When that can happen?

Reply:
Sorry for the late reply as I was traveling, It can be null when we call this [function ](https://github.com/oppia/oppia/blob/7aea401b372b38a66442e447332e4e06ca5880f6/core/templates/pages/exploration-player-page/services/audio-translation-language.service.ts#L192). This basically means we do not selected any language for audio voiceovers.

Reply:
Okay, but why do we expect `getAudioFilenamesInBfsOrder` to be called when `languageCode` is `null`. I thought it is used only when we have some lang selected.

Reply:
Yes, this function will only be called when we have selected any languageCode, but as the languageCode is of type null | string, it gives an error on assigning it to a value of type string. So, just to make sure the compiler doesn't complain about the type I am using an if statement.

Reply:
Okay, but then that means we can maybe return the empty string array at the start of this function or make sure that this function is not called with `null`, right?

Reply:
Done. We are returning an empty string array at the start of this function if languageCode is null.